cpu/registers.go

```diff
@@ -0,0 +1,75 @@
+package cpu
+
+// Register represents a typical 16-bit register.
+type Register uint16
+
+// Hi returns the most-significant 8 bits in a register.
+// For example, in the bit string representation of a register: 1111_1111_0000_0000
+// the first eight "1" bits would be returned as a uint8.
+func (r Register) Hi() uint8 {
+	return uint8(r >> 8)
+}
+
+// SetHi sets the most-significant 8 bits in the register.
+func (r *Register) SetHi(x uint8) {
+	*r = (Register(x) << 8) | Register(r.Lo())
```

Comment:
Should this be `Register(x) >> 8` to get the most-significant 8 bits in a register?

Reply:
We aren't trying to get the most-signficant 8 bits, but rather set them to the value of `x`, which is an `unsigned 8-bit integer`.

We cast it to `Register` type, and shift it left 8 bits to align them with the upper most-signficatn bits of `r`. We then `Logical Or` it with the `Lo()` bits of what `r` is already is to effectively set the upper 8bits with the contents of `x`.